messages/src/test/java/no/unit/nva/publication/messages/delete/DeleteMessageHandlerTest.java

```diff
@@ -0,0 +1,134 @@
+import no.unit.nva.publication.model.business.TicketEntry;
+import no.unit.nva.publication.model.business.UserInstance;
+import no.unit.nva.publication.service.ResourcesLocalTest;
+import no.unit.nva.publication.service.impl.MessageService;
+import no.unit.nva.publication.service.impl.ResourceService;
+import no.unit.nva.publication.service.impl.TicketService;
+import no.unit.nva.stubs.FakeContext;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.apigateway.exceptions.ApiGatewayException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.zalando.problem.Problem;
+
+class DeleteMessageHandlerTest extends ResourcesLocalTest {
+
+    private final static FakeContext CONTEXT = new FakeContext();
+    private DeleteMessageHandler handler;
+    private ByteArrayOutputStream output;
+    private ResourceService resourceService;
+    private TicketService ticketService;
+    private MessageService messageService;
+
+    @BeforeEach
+    public void setup() {
+        super.init();
+        this.resourceService = new ResourceService(client, Clock.systemDefaultZone());
+        this.ticketService = new TicketService(client);
+        this.messageService = new MessageService(client);
+        this.handler = new DeleteMessageHandler(messageService);
+        this.output = new ByteArrayOutputStream();
+    }
+
+    @Test
+    void shouldReturnNotFoundWhenMessageToDeleteDoesNotExist() throws IOException {
+        handler.handleRequest(deleteNonExistentMessageRequest(), output, CONTEXT);
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HTTP_NOT_FOUND)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(classes = {DoiRequest.class, PublishingRequestCase.class, GeneralSupportRequest.class})
+    void shouldReturnSuccessWhenTicketIsSuccessfullyDeleted(Class<? extends TicketEntry> ticketType)
+        throws ApiGatewayException, IOException {
+        var ticket = persistRandomTicket(ticketType);
+        var message = messageService.createMessage(ticket, UserInstance.fromTicket(ticket), randomString());
+
+        handler.handleRequest(deleteMessageRequest(message), output, CONTEXT);
+
+        var deletedMessage = messageService.getMessage(UserInstance.fromMessage(message), message.getIdentifier());
+
+        assertThat(deletedMessage.getStatus(), is(equalTo(MessageStatus.DELETED)));
+        assertThat(GatewayResponse.fromOutputStream(output, Void.class).getStatusCode(), is(equalTo(HTTP_OK)));
+    }
+
+    @ParameterizedTest
+    @ValueSource(classes = {DoiRequest.class, PublishingRequestCase.class, GeneralSupportRequest.class})
+    void shouldReturnNotFoundWhenUserIsAttemptingToDeleteTicketUserDoesNotOwn(Class<? extends TicketEntry> ticketType)
```

Comment:
Is it better to return 405 Method Not Allowed or 401 Unauthorized in this case?

Reply:
Not found indikerer at ressursen ikke finnes, og hvis ressursen ikke finnes kan det tolkes som at den allerede er slettet. 

Antyder ikke Method not allowed at absolutt ingen kan slette denne ressursen?

Men enig med at Unauthorized er bedre.